### ISSUE TITLE
Images glitching black

### ISSUE THREAD

**Issue by PiGP** [kivy/kivy#6220]
### Versions

* Python: 3.7.2
* OS: Win 10
* Kivy: v1.10.1
* Kivy installation method: pip install

### Description
The glitch happens when swapping images which act as toggle button, when the app is minimalized or running in the background.

In order to change the button state, while the app wasnt the main program i was using F keys on the keyboard.

When the app starts, and as long as there is user input everything works fine, the image swaps and displays correctly. The glitch happens after the app has been running in the background for some time, then after changing button state with F keys (which worked fine before) the images turn black.

After one image turns black, it doesnt matter if the app is main program and im clicking buttons with mouse, or keyboard, they all start glitching as long as i press the button with image source same as the first one that glitched.

This is how they work/look:
https://imgur.com/kxMRGg0
https://imgur.com/KX7gHZC

And this is what they look like when the bug happens:
https://imgur.com/FSTXFxC

Im pretty sure the issue is with image, not the button, as it still works fine even when the image is glitched, so the bug is only visual.

Heres thread i made when trying to solve the issue:
https://groups.google.com/forum/#!topic/kivy-users/UbzTYa_uAM0

The walkaround proposed by tshirtman, that "fixed" the issue was changing this value timeout value:
https://github.com/kivy/kivy/blob/c4d68949b42e97de270528bdc2a39ec9393bd7f4/kivy/core/image/__init__.py#L88

The app im running has 4 threads running in the background, and the main function is a while loop thats refreshing every 0.01s, there are around 10 if statements checking for keyboard input, and if it occurs then they activate methods in classes (one of which is button swap method inside current screenmanager class)

I hope it helps, cant really give a proper feedback because im just a beginner. 

### Code and Logs

```python
def main():
    while True:
        if win32api.GetAsyncKeyState(F4):
            Class.method1()
        if win32api.GetAsyncKeyState(F5):
            Class.method2()
        if win32api.GetAsyncKeyState(F6):
            Class.method3()
        if win32api.GetAsyncKeyState(F7):
            Class.method4()

class Class():
    value1 = False
    value2 = False
    value3 = False
    value4 = False

    def method1():
        #do something
        value1 = not value1
        if value1 == True:
            #do something
            self.ids.btn1.source = './images/Dashboard/btn_on.png'
        else:
            #do something
            self.ids.btn1.source ='./images/Dashboard/btn_off.png'

    def method2():
        #do something
        value2 = not value2
        if value2 == True:
            #do something
            self.ids.btn2.source = './images/Dashboard/btn_on.png'
        else:
            #do something
            self.ids.btn2.source ='./images/Dashboard/btn_off.png'


#AND THEN THE BUTTON IN .KV FILE:
		Label:
			id: btn1label
			size_hint: (.1,.1)
			pos_hint: {"x":0.03, "top":0.65}
			Image:
				id: btn1
				source: './images/Dashboard/btn_off.png'
				y: self.parent.y
				x: self.parent.x
				size: 40, 40
				#allow_stretch: True

			Button:
				size:80,35
				background_color: 0,0,0,0
				y: self.parent.y+23
				x: self.parent.x
				on_press: root.method1()


#AND THE LOGS WHEN EVERYTHING WORKS FINE:
TRACE:kivy:[Context     ] releasing 1 textures: array('I', [33])
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C488368>, key=texture, value=<code object <module> at 0x0000016C0EFB7270, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 123>, 'self.texture'
TRACE:kivy:[Lang        ] call_fn => value=<Texture hash=1563878309960 id=32 size=(200, 92) colorfmt='rgba' bufferfmt='ubyte' source='False|C:\\Users\\Irrelevant\\Desktop\\UI KIvy\\FirstTry\\images\\Dashboard\\btn_on.png|0|0' observers=0>
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C488368>, key=pos, value=<code object <module> at 0x0000016C0EFB7390, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 125>, 'self.center_x - self.norm_image_size[0] / 2., self.center_y - self.norm_image_size[1] / 2.'
TRACE:kivy:[Lang        ] call_fn => value=(24.0, 228.60000000000002)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C488368>, key=size, value=<code object <module> at 0x0000016C0EFB7300, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 124>, 'self.norm_image_size'
TRACE:kivy:[Lang        ] call_fn => value=(80, 36.800000000000004)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C59E7C8>, key=pos, value=<code object <module> at 0x0000016C0EFA4810, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 10>, 'int(self.center_x - self.texture_size[0] / 2.), int(self.center_y - self.texture_size[1] / 2.)'
TRACE:kivy:[Lang        ] call_fn => value=(702, 4)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C59E7C8>, key=size, value=<code object <module> at 0x0000016C0EFA4780, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 9>, 'self.texture_size'
TRACE:kivy:[Lang        ] call_fn => value=[68, 19]
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C59E7C8>, key=texture, value=<code object <module> at 0x0000016C0EFA46F0, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 8>, 'self.texture'
TRACE:kivy:[Lang        ] call_fn => value=<Texture hash=1563878310656 id=0 size=(68, 19) colorfmt='rgba' bufferfmt='ubyte' source=None observers=1>
TRACE:kivy:[Context     ] releasing 1 textures: array('I', [30])
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C59E7C8>, key=pos, value=<code object <module> at 0x0000016C0EFA4810, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 10>, 'int(self.center_x - self.texture_size[0] / 2.), int(self.center_y - self.texture_size[1] / 2.)'
TRACE:kivy:[Lang        ] call_fn => value=(698, 4)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C59E7C8>, key=size, value=<code object <module> at 0x0000016C0EFA4780, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 9>, 'self.texture_size'
TRACE:kivy:[Lang        ] call_fn => value=[75, 19]
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C59E7C8>, key=texture, value=<code object <module> at 0x0000016C0EFA46F0, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 8>, 'self.texture'
TRACE:kivy:[Lang        ] call_fn => value=<Texture hash=1563823738104 id=0 size=(75, 19) colorfmt='rgba' bufferfmt='ubyte' source=None observers=1>
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C488368>, key=texture, value=<code object <module> at 0x0000016C0EFB7270, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 123>, 'self.texture'
TRACE:kivy:[Lang        ] call_fn => value=<Texture hash=1563823737408 id=9 size=(256, 118) colorfmt='rgba' bufferfmt='ubyte' source='False|C:\\Users\\Irrelevant\\Desktop\\UI KIvy\\FirstTry\\images\\Dashboard\\btn_off.png|0|0' observers=0>
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C488368>, key=pos, value=<code object <module> at 0x0000016C0EFB7390, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 125>, 'self.center_x - self.norm_image_size[0] / 2., self.center_y - self.norm_image_size[1] / 2.'
TRACE:kivy:[Lang        ] call_fn => value=(24.0, 228.56250000000003)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C488368>, key=size, value=<code object <module> at 0x0000016C0EFB7300, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 124>, 'self.norm_image_size'

#AND THE LOGS WHEN THE IMAGES START TO GLITCH:
TRACE:kivy:[Context     ] releasing 1 textures: array('I', [30])
TRACE:kivy:[Context     ] releasing 6 textures: array('I', [11, 12, 13, 14, 15, 16])
TRACE:kivy:[Context     ] releasing 3 textures: array('I', [17, 18, 20])
TRACE:kivy:[Context     ] releasing 6 textures: array('I', [21, 22, 23, 24, 25, 26])
DEBUG:kivy:[ImageSDL2   ] Load <C:\Users\Irrelevant\Desktop\UI KIvy\FirstTry\images\Dashboard\btn_off.png>
TRACE:kivy:[Image       ] 'C:\\Users\\Irrelevant\\Desktop\\UI KIvy\\FirstTry\\images\\Dashboard\\btn_off.png', populate to textures (1)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C488728>, key=texture, value=<code object <module> at 0x0000016C0EFB7270, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 123>, 'self.texture'
TRACE:kivy:[Lang        ] call_fn => value=<Texture hash=1563844570592 id=0 size=(256, 118) colorfmt='rgba' bufferfmt='ubyte' source='False|C:\\Users\\Irrelevant\\Desktop\\UI KIvy\\FirstTry\\images\\Dashboard\\btn_off.png|0|0' observers=0>
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C488728>, key=pos, value=<code object <module> at 0x0000016C0EFB7390, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 125>, 'self.center_x - self.norm_image_size[0] / 2., self.center_y - self.norm_image_size[1] / 2.'
TRACE:kivy:[Lang        ] call_fn => value=(24.0, 188.0625)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C488728>, key=size, value=<code object <module> at 0x0000016C0EFB7300, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 124>, 'self.norm_image_size'
TRACE:kivy:[Lang        ] call_fn => value=(80, 36.875)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C59E7C8>, key=pos, value=<code object <module> at 0x0000016C0EFA4810, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 10>, 'int(self.center_x - self.texture_size[0] / 2.), int(self.center_y - self.texture_size[1] / 2.)'
TRACE:kivy:[Lang        ] call_fn => value=(691, 4)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C59E7C8>, key=size, value=<code object <module> at 0x0000016C0EFA4780, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 9>, 'self.texture_size'
TRACE:kivy:[Lang        ] call_fn => value=[90, 19]
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C59E7C8>, key=texture, value=<code object <module> at 0x0000016C0EFA46F0, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 8>, 'self.texture'
TRACE:kivy:[Lang        ] call_fn => value=<Texture hash=1563844570360 id=0 size=(90, 19) colorfmt='rgba' bufferfmt='ubyte' source=None observers=1>
TRACE:kivy:[Context     ] releasing 1 textures: array('I', [33])
DEBUG:kivy:[ImageSDL2   ] Load <C:\Users\Irrelevant\Desktop\UI KIvy\FirstTry\images\Dashboard\btn_on.png>
TRACE:kivy:[Image       ] 'C:\\Users\\Irrelevant\\Desktop\\UI KIvy\\FirstTry\\images\\Dashboard\\btn_on.png', populate to textures (1)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C488728>, key=texture, value=<code object <module> at 0x0000016C0EFB7270, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 123>, 'self.texture'
TRACE:kivy:[Lang        ] call_fn => value=<Texture hash=1563844570128 id=0 size=(200, 92) colorfmt='rgba' bufferfmt='ubyte' source='False|C:\\Users\\Irrelevant\\Desktop\\UI KIvy\\FirstTry\\images\\Dashboard\\btn_on.png|0|0' observers=0>
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C488728>, key=pos, value=<code object <module> at 0x0000016C0EFB7390, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 125>, 'self.center_x - self.norm_image_size[0] / 2., self.center_y - self.norm_image_size[1] / 2.'
TRACE:kivy:[Lang        ] call_fn => value=(24.0, 188.1)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C488728>, key=size, value=<code object <module> at 0x0000016C0EFB7300, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 124>, 'self.norm_image_size'
TRACE:kivy:[Lang        ] call_fn => value=(80, 36.800000000000004)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C59E7C8>, key=pos, value=<code object <module> at 0x0000016C0EFA4810, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 10>, 'int(self.center_x - self.texture_size[0] / 2.), int(self.center_y - self.texture_size[1] / 2.)'
TRACE:kivy:[Lang        ] call_fn => value=(694, 4)
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C59E7C8>, key=size, value=<code object <module> at 0x0000016C0EFA4780, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 9>, 'self.texture_size'
TRACE:kivy:[Lang        ] call_fn => value=[83, 19]
TRACE:kivy:[Lang        ] call_fn <kivy.graphics.vertex_instructions.Rectangle object at 0x0000016C1C59E7C8>, key=texture, value=<code object <module> at 0x0000016C0EFA46F0, file "D:\Anaconda\lib\site-packages\kivy\data\style.kv", line 8>, 'self.texture'
TRACE:kivy:[Lang        ] call_fn => value=<Texture hash=1563844569896 id=0 size=(83, 19) colorfmt='rgba' bufferfmt='ubyte' source=None observers=1>

```

**Comment by PiGP** [comment]
By the looks of logs, this line in particular:
DEBUG:kivy:[ImageSDL2   ] Load <C:\Users\Irrelevant\Desktop\UI KIvy\FirstTry\images\Dashboard\btn_on.png>

It might seem like ImageSDL2 is causing the issue(at least to me, but im a noob), but after disabling it in config, i get this log when the glitch happens:

DEBUG:kivy:[ImagePygame ] Load <C:\Users\Irrelevant\Desktop\UI KIvy\FirstTry\images\Dashboard\btn_on.png>

**Comment by ElliotGarbus** [comment]
@Julian-O 
There is not enough content here to reproduce the issue.  I recommend closing.

**Comment by Julian-O** [comment]
Wait... I think I know what is happening here! It affects one of my Kivy apps on Android, and I located what I think was the cause, but haven't fixed it yet.

> source: './images/Dashboard/btn_off.png'

This is a relative path to the current working directory.

The image is loaded when the program first runs (and put in the cache), but eventually is removed to save memory.
The next time it needs to be displayed, it goes to storage again... but the current working directory has changed and it can't find the file.

Replace the relative paths with with full paths.

Closing as not a bug in Kivy.
